src/config.js

```diff
@@ -33,31 +31,62 @@ export function applyConfigToArgv({
         `specified in camel case: "${camelCase(option)}"`);
     }
 
-    if (option === commandExecuted) {
+    if (typeof options[option] === 'object' &&
+      typeof configObject[option] === 'object') {
+      // Descend into the nested configuration for a sub-command.
       newArgv = applyConfigToArgv({
         argv,
-        configObject: configObject[commandExecuted],
-        defaultValues: defaultValues[commandExecuted],
+        configObject: configObject[option],
+        options: options[option],
         configFileName});
       continue;
     }
 
+    const decamelizedOptName = decamelize(option, '-');
+
+    if (typeof options[decamelizedOptName] !== 'object') {
+      throw new UsageError(`The config file at ${configFileName} specified ` +
+        `an unknown option: "${option}"`);
+    }
+    if (options[decamelizedOptName].type === undefined) {
+      throw new WebExtError(
+        `Option: ${option} was defined without a type.`);
+    }
+
+    const expectedType = options[decamelizedOptName].type ===
+      'count' ? 'number' : options[decamelizedOptName].type;
+
+    if (configObject[option]) {
+      const optionType = typeof(configObject[option]);
+      if (optionType !== expectedType &&
+        expectedType !== undefined) {
+        throw new UsageError(`The config file at ${configFileName} specified ` +
+          `the type of "${option}" incorrectly as "${optionType}"` +
+          ` (expected type: "${expectedType}")`);
+      }
+    }
+
+    let defaultValue;
+    if (options[decamelizedOptName]) {
+      if (options[decamelizedOptName].default !== undefined) {
```

Comment:
I don't understand the need for switching these. You mentioned that `true` values will never be honored but I don't see how because the condition is `!== undefined`. Could you elaborate a little?